tests/column_aggregate_expectations_distributional/test_pandas_dataset_distributional_expectations.py

```diff
@@ -580,6 +596,208 @@ def test_expect_column_kl_divergence_to_be_less_than_infinite_observed_value(sel
+                                "bins": [-2,-1,0,1,2],
+                                "weights": [3/15.0, 1/15.0, 2/15.0, 4/15.0],
+                                "tail_weights": [2/15.0,3/15.0]
+                            },
+                            "expected_partition": {
+                                "bins": [-2,-1,0,1,2],
+                                "weights": [0.15*0.95,0.05*0.95,0.05*0.95,0.45*0.95],
+                                "tail_weights":[0.3*0.95,0.05]
+                            }
+                        }
+                            
+        actual_details=continuous_df.expect_column_kl_divergence_to_be_less_than("x",
+                                                                                 continuous_partition_object,
+                                                                                 0.05,
+                                                                                 tail_weight_holdout=0.05, 
+                                                                                 internal_weight_holdout=0,
+                                                                                 result_format="COMPLETE")["result"]["details"]
+        assertDeepAlmostEqual(expected_details,actual_details)
+        
+        
+        #Bounded Endpoints, non-zero tail holdout, non-zero internal holdout
+        continuous_partition_object={"weights":[0.3,0.15,0.0,0.10,0.45],
+            "bins":[-3,-2,-1,0,1,2]}
+        
+        
+        expected_details= {
+                            "observed_partition": {
+                                # return expected_bins, since we used those bins to compute the observed_weights
+                                "bins": [-3,-2,-1,0,1,2],
+                                "weights": [2/15.0,3/15.0, 1/15.0, 2/15.0, 4/15.0],
+                                "tail_weights": [0.0,3/15.0]
+                            },
+                            "expected_partition": {
+                                "bins": [-3,-2,-1,0,1,2],
+                                "weights": [0.3*0.9,0.15*0.9,0.05,0.10*0.9,0.45*0.9],
+                                "tail_weights":[0.05/2,0.05/2]
+                            }
+                        }
+                            
+        actual_details=continuous_df.expect_column_kl_divergence_to_be_less_than("x",
+                                                                                 continuous_partition_object,
+                                                                                 0.05,
+                                                                                 tail_weight_holdout=0.05, 
+                                                                                 internal_weight_holdout=0.05,
+                                                                                 result_format="COMPLETE")["result"]["details"]
+        
+        assertDeepAlmostEqual(expected_details,actual_details)
```

Comment:
Could you add a check that the newly returned partition objects (now including the tail_weights) are correctly identified as valid by `util.is_valid_continuous_partition_object`?

Reply:
Yes, though the way it is currently written, `util.is_valid_continuous_partition_object` will consider the returned partitions as invalid. The returned partition objects in my rewrite do not have `weights` that will sum to 1; `weights+tail_weights` will have entries that sum to 1. I did not rewrite `util.is_valid_continuous_partition_object` to consider the returned partition objects; it only considers input partition objects.  Since the user is not expected to give an input partition object with tail weights,  `util.is_valid_continuous_partition_object` doesn't expect tail weights. 

Are we expecting users to give partition objects with `tail_weights`? If so, the easiest thing would be to just rewrite  `util.is_valid_continuous_partition_object`  to expect `tail_weights`. If we don't want to change how users input partition objects, I will just manipulate the returned partition objects in the test program so that they will be valid for `util.is_valid_continuous_partition_object`. We could also rewrite `util.is_valid_continuous_partition_object` to distinguish between returned and input partitions. Do any of those seem better than the others?

I lean towards keeping the input expectations from the user the same (i.e. no tail weights from users) and rewrite  `util.is_valid_continuous_partition_object` to distinguish between returned and input partitions.

Reply:
I agree with the bottom line--whether or not the expectation is changed to allow users to include tail weights, I think it makes more sense to update `is_valid_continuous_partition_object` to be aware of tail_weights. 

I also do think it makes sense to allow users to give partition objects that include tail_weights, and if they do, it should behave fundamentally they same as if they specified a `tail_weight_holdout` (i.e. you should be able to specify only one or the other, having both should be an error). Does that make sense?

Reply:
That makes sense. I will plan to work on this this weekend. Hopefully, I can get out a new pull request before next week. Thanks!